acados/ocp_nlp/ocp_nlp_reg_common.c

```diff
@@ -220,5 +249,32 @@ void acados_project(int dim, double *A, double *V, double *d, double *e, double
 }
 
 
+void acados_project_adaptive_eps(int dim, double *A, double *V, double *d, double *e, double max_cond_block)
+{
+    int i;
+    acados_eigen_decomposition(dim, A, V, d, e);
+    double max_eig = 0.0;
+    double eps;
+    int eig_0_count = 0;
+
+    // compute max and min eigenvalues
+    for (i=0; i < dim; i++)
+    {
+        max_eig = MAX(max_eig, d[i]); // negative eigenvalues are clipped, i.e., not important for condition number
+        if (fabs(d[i]) == 0.0)
+            eig_0_count += 1;
+    }
+    if (eig_0_count == dim)
```

Comment:
Instead of counting the zero eigenvalues you could Check whether `max_eig` is zero. Then, the method also works if all eigenvalues are negative :)